packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

```diff
@@ -198,9 +201,175 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
+                }
+
+                const line = this.createResultLineNode(result, newFileNode);
+                newFileNode.children.push(line);
+                rootFolderNode.children.push(newFileNode);
+            }
+
+        } else {
+            const newRootFolderNode = this.createRootFolderNode(result.root);
+            tree.set(result.root, newRootFolderNode);
+            const newFileNode = this.createFileNode(result.root, name, path, result.fileUri, newRootFolderNode);
+            this.collapseFileNode(newFileNode, collapseValue);
+            if (isDirtyResult) {
+                this.dirtyFileUris.add(newFileNode.fileUri);
+            }
+
+            const line = this.createResultLineNode(result, newFileNode);
+            newFileNode.children.push(line);
+            newRootFolderNode.children.push(newFileNode);
+        }
+    }
+
+    /**
+     * Search in all dirty editors in the current workspace.
+     * @returns The number of match results in dirty files.
+     */
+    searchInDirtyFiles(searchTerm: string, searchOptions: SearchInWorkspaceOptions): number {
+        // Get all dirty editor widgets.
+        let dirtyWidgets: EditorWidget[] = this.editorManager.all.filter(w => w.saveable.dirty);
+        // Filter to exclude dirty widgets that should be ignored.
+        if (!searchOptions.includeIgnored) {
+            const ignoredPatterns = this.fileSystemPreferences.get('files.exclude');
+            dirtyWidgets = dirtyWidgets.filter(widget => {
+                for (const pattern in ignoredPatterns) {
+                    if (ignoredPatterns[pattern] && minimatch(widget.title.label, pattern)) {
+                        return false;
+                    }
+                }
+                return true;
+            });
+        }
+        // Filter to only search dirty widgets in `files to include`.
+        if (searchOptions.include && searchOptions.include.length > 0) {
+            const includedPatterns: string[] = searchOptions.include;
+            dirtyWidgets = dirtyWidgets.filter(widget => includedPatterns.some(pattern => minimatch(widget.title.label, pattern)));
+        }
+        // Filter to only search dirty widgets that are not in `files to exclude`.
+        if (searchOptions.exclude && searchOptions.exclude.length) {
+            const excludedPatterns: string[] = searchOptions.exclude;
+            dirtyWidgets = dirtyWidgets.filter(widget => !excludedPatterns.some(pattern => minimatch(widget.title.label, pattern)));
+        }
+
+        let numberOfResults = 0;
+        dirtyWidgets.forEach(async w => {
+            const fileUri: string = w.editor.uri.toString();
+            const roots = await this.workspaceService.roots;
+            const root: string = this.getRoot(w.editor.uri.path.toString(), roots.map(r => r.uri)).toString();
+            const fileContent: string = w.editor.document.getText();
+            const matches = this.findMatches(searchTerm, fileContent, searchOptions);
```

Comment:
monaco model has API to search matches: https://github.com/microsoft/monaco-editor/blob/0a4ed9a829ce6deaf794ea3dfaf4e814a4c10120/monaco.d.ts#L1646-L1669

You don't need to implement it.

Reply:
That sounds much better, I assume it requires waiting on #5901

Reply:
no, it should be there in Monaco 0.14 as well